src/WebJobs.Script.Extensibility/ScriptBindingContext.cs

```diff
@@ -107,12 +114,32 @@ public ScriptBindingContext(JObject bindingMetadata)
         public TValue GetMetadataValue<TValue>(string name, TValue defaultValue = default(TValue))
         {
             JToken value = null;
-            if (Metadata.TryGetValue(name, StringComparison.OrdinalIgnoreCase, out value))
+            try
             {
-                return value.Value<TValue>();
+                if (Metadata.TryGetValue(name, StringComparison.OrdinalIgnoreCase, out value))
+                {
+                    return value.Value<TValue>();
+                }
+            }
+            catch (FormatException e)
+            {
+                throw new FormatException($"Invalid value specified for binding property '{name}' of type {PrettyTypeName(typeof(TValue))}.", e);
             }
 
             return defaultValue;
         }
+
+        private static string PrettyTypeName(Type t)
```

Comment:
If I'm developing a non-C# function I'd prefer to see `Nullable<Int32>` over `System.Nullable<System.Int32>`

I don't disagree with your points, I just don't think that naming conflicts will be a frequent issue and I like brevity (to a fault, probably)